tests/network_manager_test.go

```diff
@@ -36,3 +41,72 @@ func TestDataURLSkipRequest(t *testing.T) {
 
 	assert.True(t, lc.contains("skipped request handling of data URL"))
 }
+
+func TestMetricsEmission(t *testing.T) {
+	tb := newTestBrowser(t, withHTTPServer())
+
+	url := tb.URL("/get")
+	browserLoadedTags := map[string]string{
+		"group": "",
+		"url":   "about:blank",
+	}
+	browserTags := map[string]string{
+		"group": "",
+		"url":   url,
+	}
+	httpTags := map[string]string{
+		"method":              "GET",
+		"url":                 url,
+		"status":              "200",
+		"group":               "",
+		"proto":               "http/1.1",
+		"from_cache":          "false",
+		"from_prefetch_cache": "false",
+		"from_service_worker": "false",
+	}
+	expMetricTags := map[string]map[string]string{
+		common.BrowserDOMContentLoaded.Name:     browserLoadedTags,
+		common.BrowserLoaded.Name:               browserLoadedTags,
+		common.BrowserFirstPaint.Name:           browserTags,
+		common.BrowserFirstContentfulPaint.Name: browserTags,
+		k6metrics.DataSentName: map[string]string{
+			"group":  "",
+			"method": "GET",
+			"url":    url,
+		},
+		k6metrics.HTTPReqsName:              httpTags,
+		k6metrics.HTTPReqDurationName:       httpTags,
+		k6metrics.DataReceivedName:          httpTags,
+		k6metrics.HTTPReqConnectingName:     httpTags,
+		k6metrics.HTTPReqTLSHandshakingName: httpTags,
+		k6metrics.HTTPReqSendingName:        httpTags,
+		k6metrics.HTTPReqReceivingName:      httpTags,
+	}
+
+	p := tb.NewPage(nil)
+	resp := p.Goto(url, tb.rt.ToValue(struct {
+		WaitUntil string `js:"waitUntil"`
+	}{WaitUntil: "networkidle"}))
+	require.NotNil(t, resp)
+
+	// TODO: Remove this sleep. It's only needed to wait for all metrics to be
+	// emitted, but this should be synchronized with waitUntil
+	// load/networkidle/domcontentloaded. Without this the test would be flaky.
+	time.Sleep(100 * time.Millisecond)
```

Comment:
Maybe you can this one to check whether the network is idle/waitUntil/load/etc:

https://github.com/grafana/xk6-browser/blob/758d14bb6082e0c8bbea7933522f7f84cff985fc/common/frame.go#L379-L384

It's because metrics are emitted on lifecycle events:

https://github.com/grafana/xk6-browser/blob/758d14bb6082e0c8bbea7933522f7f84cff985fc/common/frame_session.go#L687-L766

Reply:
Good point, I thought `waitUntil` would accomplish the same thing.

Resolved by 3668b38. It seems stable from my local tests.

Reply:
Actually, yeah, that's weird and it might even be a bug. A new issue? @imiric